api-server/src/server/component-passport.js

```diff
@@ -85,13 +86,22 @@ export const devSaveResponseAuthCookies = () => {
 
 export const devLoginRedirect = () => {
   return (req, res) => {
-    // this mirrors the production approach, but without any validation
+    // this mirrors the production approach, but only validates the prefix
     let { returnTo, origin, pathPrefix } = getRedirectParams(
       req,
-      params => params
+      ({ returnTo, origin, pathPrefix }) => {
+        pathPrefix = availableLangs.client.includes(pathPrefix)
+          ? pathPrefix
+          : '';
+        return {
+          returnTo,
+          origin,
+          pathPrefix
+        };
+      }
     );
     returnTo += isRootPath(getRedirectBase(origin, pathPrefix), returnTo)
-      ? 'learn'
+      ? '/learn'
```

Comment:
Maybe it is GitPod, but this change causes:
```
https://8000-emerald-capybara-b8ln1x5v.ws-eu08.gitpod.io//learn
```

Reply:
Could you provide steps to reproduce?  I can't replicate the bug in Gitpod.

Reply:
Add this:
```javascript
returnTo += isRootPath(getRedirectBase(origin, pathPrefix), returnTo)
      ? '/learn'
      : '';
console.log(returnTo);
```
2) Navigate to the main landing page
3) Sign out
4) Sign in
5) Look in server console

Reply:
On `.dev`:
```console
Navigated to https://www.freecodecamp.dev//learn?messages=success%5B0%5D%3Dflash.signin-success
```

Reply:
It does not break anything, but it still concerns me:
```
Request URL: https://www.freecodecamp.dev//learn?messages=success%5B0%5D%3Dflash.signin-success
```

Reply:
For context: we discussed this in chat and realised that this behaviour is nothing to do with this PR.